### PR TITLE
Fix PG string escaping behavior

### PR DESCRIPTION
**Pull request by tgriesser**
Reverts to pre 0.11 behavior for custom parameters, #1602 and adds more consistent escaping for postgresql queries: #1548

Also re-arranges where some of the query parsing/parameterization is happening internally in raw queries for consistency.

This will land in 0.12 release

### CODE REVIEW

File: src/query/string.js [hunk 1]
```diff
@@ -1,134 +1,161 @@
...
+  const finalEscapeDate = config.escapeDate || dateToString
+  const finalEscapeArray = config.escapeArray || arrayToList
+  const finalEscapeBuffer = config.escapeBuffer || bufferToString
+  const finalEscapeString = config.escapeString || escapeString
+  const finalEscapeObject = config.escapeObject || escapeObject
+  const finalWrap = config.wrap || wrapEscape
+
+  function escapeFn(val, finalEscape, ctx) {
+    if (val === undefined || val === null) {
+      return 'NULL';
```
Comment (wubzz) [inline src/query/string.js]: @tgriesser A lot of changes so hard to follow up, but just to verify, is this part run *before* or *after* the checks run during compile such as https://github.com/tgriesser/knex/pull/1661/files#diff-2fe557044fa519ea01fdf735c4846284R99 ?

If run before, then these checks would never work, so just double-checking. In any case,  :+1: :+1: for this pr.

Comment (tgriesser) [inline src/query/string.js]: They run after, basically if `.toString` is used, this is the escaping that happens
